lib/kernel/src/inet_db.erl

```diff
@@ -1800,17 +1793,18 @@ delete_older(CacheDb, TM, N) ->
 
 delete_older('$end_of_table', _, _, _, M) ->
     M;
-delete_older(_, _, _, N, M) when N =< M ->
+delete_older(_, _, _, M, M) ->
     M;
-delete_older(Domain, CacheDb, TM, N, M) ->
-    Next = ets:next(CacheDb, Domain),
-    Fun = fun(RR, MM) when RR#dns_rr.cnt =< TM ->
-		  ets:match_delete(CacheDb, RR),
-		  MM + 1;
-	     (_, MM) ->
-		  MM
+delete_older(Key, CacheDb, TM, N, M) ->
+    Next = ets:next(CacheDb, Key),
+    M1 =
+	case ets:lookup(CacheDb, Key) of
+	    [{_,RR}] when RR#dns_rr.cnt =< TM ->
+		ets:delete(CacheDb, Key),
+		M + 1;
+	    _ ->
+		M
```

Comment:
These lines are not covered in our daily tests by running all test suites in Kernel.